add a warning if we think istio-proxy injection is causing problems (#3545)
apollographql/router: docs/source/containerization/kubernetes.mdx

curl "http://localhost:8088/health"

## Using `istio` with the router

> The [istio service mesh](https://istio.io/) is a very popular choice for enhanced traffic routing within kubernetes.

Reviewer: ```suggestion
The [istio service mesh](https://istio.io/) is a very popular choice for enhanced traffic routing within Kubernetes.
```

> We have encountered an [issue](https://github.com/apollographql/router/issues/3533) with `istio-proxy` pod injection. It is possible for the router to start executing at the same time that istio is reconfiguring networking for the router pod. This is an issue with `istio`, not the router, and the fix is to follow the istio advice documented [here](https://istio.io/latest/docs/ops/common-problems/injection/#pod-or-containers-start-with-network-issues-if-istio-proxy-is-not-ready).

Reviewer: ```suggestion
`istio-proxy` pod injection can cause an [issue](https://github.com/apollographql/router/issues/3533) in the router. The router may start executing at the same time that istio is reconfiguring networking for the router pod. This is an issue with `istio`, not the router, and you can resolve it by following the advice in [istio's injection documentation](https://istio.io/latest/docs/ops/common-problems/injection/#pod-or-containers-start-with-network-issues-if-istio-proxy-is-not-ready).
```

Always good to use descriptive link text: https://developers.google.com/style/link-text
